test/test_unistd.rs

```diff
@@ -404,3 +404,22 @@ fn test_pipe2() {
     let f1 = FdFlag::from_bits_truncate(fcntl(fd1, FcntlArg::F_GETFD).unwrap());
     assert!(f1.contains(FdFlag::FD_CLOEXEC));
 }
+
+#[test]
+fn test_gethostname() {
```

Comment:
Please put the addition into a separate commit before the rest of these changes.